### PR TITLE
Add in-memory storage DI, node sha256 DI, and common tests

### LINKED ISSUE #223: Update common tests with DI
Some of the tests in `src/common/` rely on mocking. In some cases, we may be able to replace mocking with dependency injection.

We can update:

- `cid-log.test.ts` on `bgins/testing` which mocks `localforage`

We cannot update:

- `api.test.ts` because it uses `fetch` and we do not currently have DI for network requests

### PR DESCRIPTION
**Pull request by bgins**
## Summary

This PR fixes/implements the following **bugs/features**

* [x] Add in-memory storage DI implementation
* [x] Add node WebCrypto sha256 implementation
* [x] Add cid-log tests
* [x] Add async race test
* [x] Add identifiers tests

This PR adds an in-memory storage implementation for use in Jest. The implementation mostly matches the `localforage` API, though it does not currently restrict to the types that `localforage` can store.

We are also adding tests for `cid-log.ts`, `async.ts`, and `identifiers.ts` in the `src/common/` directory. Note that the `cid-log` tests use the new storage DI implementation and the `identifiers` tests use the sha256 implementation.

## Test plan (required)

Run `npm run test:unit`. All tests should pass.

## Related issues

Closes #223 
Part of https://github.com/fission-suite/webnative/issues/201

## After Merge
* [ ] Does this change invalidate any docs or tutorials? _If so ensure the changes needed are either made or recorded_
* [ ] Does this change require a release to be made? Is so please create and deploy the release

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -17,6 +17,7 @@ module.exports = {
       },
     }],
     "@typescript-eslint/no-use-before-define": ["off"],
-    "@typescript-eslint/semi": ["error", "never"]
+    "@typescript-eslint/semi": ["error", "never"],
+    "@typescript-eslint/ban-ts-comment": 1
```
Comment (matheus23) [inline .eslintrc.js:21]: What does the value `1` mean here? Online I found that you can set this to `"off"` and that turns it off?

File: src/common/cid-log.test.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
+import * as fc from 'fast-check'
+import CID from 'cids'
+import multihash from 'multihashing-async'
+import * as cidLog from './cid-log'
+import * as storage from '../storage'
+
+async function generateCids(data: Uint8Array[]): Promise<string[]> {
+  const promisedCids = data.map(async bytes => {
+    const mhash = await multihash(bytes, 'sha2-256')
```
Comment (matheus23) [inline src/common/cid-log.test.ts:9]: Oh my god. I needed this in one of my tests haha.

File: src/common/async.test.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import * as fc from 'fast-check'
+import { race } from './async'
```
Comment (matheus23) [inline src/common/async.test.ts:2]: Wait. Why do we have our own implementation for [`Promise.race`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/race)?

No action required immediately, but I think we should switch that out for the native variant. No need to test this ourselves then :+1:

**Review by matheus23 — APPROVED** [review]
:pray: 
:raised_hands: 
:+1:

File: .eslintrc.js (hunk 1 above)
Comment (bgins) [reply .eslintrc.js:21]: `0` means off, `1` means warn, and `2` means error. We might be able to `@typescript-eslint/eslint-plugin` to a newer version to get something a bit more explicit. The newer interface is `bool` or `"allow-with-description"` that makes you write down why.

File: src/common/cid-log.test.ts (hunk 2 above)
Comment (bgins) [reply src/common/cid-log.test.ts:9]: Yeah! That's been super helpful for getting some real CIDs :smile:

File: src/common/async.test.ts (hunk 3 above)
Comment (bgins) [reply src/common/async.test.ts:2]: Great point, I was also noticing that Jest coverage doesn't think we are actually calling `promise.then(resolve, reject)` in the implementation. (I tested it with logging and we are, but it's not ideal that it doesn't show up in coverage.)

Comment (walkah) [reply src/common/async.test.ts:2]: TIL! I didn't realize Promises has `.race` built in.

**Review by walkah — APPROVED** [review]
30 rock college - DOxlkyuDxTrna

*test coverage intensifies*

**Pull request merged**